<?xml version="1.0"?>
<launch>

<include file ="$(find a_robot_platform)/launch/map_server.launch"/>

<node  name="amcl_location" pkg="a_robot_platform" type="main_global_location" output="screen">
	<param name="use_map_topic" value="true"/>
	<param name="first_map_only" value="true"/>
	
	<param name="gui_publish_rate" value="-1.0"/>
	<param name="save_pose_rate" value="0.5"/>
	<param name="laser_min_range" value="0.10"/>
	<param name="laser_max_range" value="3.0"/>
	<param name="laser_max_beams" value="360"/>
	<param name="min_particles" value="400"/>
	<param name="max_particles" value="3000"/>
	<param name="kld_err" value="0.05"/>
	<param name="kld_z" value="0.99"/>
	<param name="odom_alpha1" value="0.5"/>
	<param name="odom_alpha2" value="0.5"/>
	<param name="odom_alpha3" value="0.2"/>
	<param name="odom_alpha4" value="0.2"/>
	<param name="odom_alpha5" value="0.1"/>
	
        <param name="initial_pose_x" value="0.0"/>
        <param name="initial_pose_y" value="0.0"/>
        <param name="initial_pose_z" value="0.0"/>
        <param name="initial_cov_xx" value="0.25"/>
        <param name="initial_cov_yy" value="0.25"/>
        <param name="initial_cov_aa" value="0.068"/>

	<param name="do_beamskip" value="false"/>
	<param name="beam_skip_distance" value="0.5"/>
	<param name="beam_skip_threshold" value="0.3"/>
	<param name="beam_skip_error_threshold_" value="0.9"/>
	<param name="laser_z_hit" value="0.5"/>
	<param name="laser_z_rand" value="0.5"/>
	<param name="laser_z_short" value="0.05"/>
	<param name="laser_z_max" value="0.05"/>
	<param name="laser_sigma_hit" value="0.2"/>
	<param name="laser_lambda_short" value="0.1"/>
	<param name="laser_likelihood_max_dist" value="2.0"/>
	<param name="laser_model_type" value="likelihood_field"/>
	<param name="odom_model_type" value="diff"/>
	<param name="update_min_d" value="0.005"/>
	<param name="update_min_a" value="0.01"/>
	<param name="resample_interval" value="1"/>
	<param name="transform_tolerance" value="0.2"/>
	<param name="recovery_alpha_slow" value="0.0"/>
	<param name="recovery_alpha_fast" value="0.0"/>
	<param name="tf_broadcast" value="true"/>
	<param name="odom_frame_id" value="odom"/>
	<param name="base_frame_id" value="base_link"/>
	<param name="global_frame_id" value="map"/>

        <param name="pose_diff" value="0.025"/>
        <param name="angle_diff" value="0.05"/>
        <param name="map_depth_num" value="2"/>
        <param name="publish_scan" value="true"/>
        <param name="write_pose" value="false"/>
 　　　　　　　<param name="max_itera" value="6"/>
        <param name="add_close_loop" value="true"/>
 　　　　　　　<param name="use_amcl_pose" value="false"/>

	<param name="subMap_resolution" value="0.1"/>
        <param name="subMap_width" value="40"/>
 　　　　　　　<param name="subMap_height" value="40"/>
	<param name="submapFilter" value="3"/>
　　　　　　　　<param name ="submapExpandCnt" value="3"/>

</node>
</launch>
